crates/runix/src/flake_ref/indirect.rs

```diff
@@ -33,6 +44,16 @@ impl TryFrom<Attrs> for IndirectRef {
         };
         let id = id.clone();
         let mut attributes = BTreeMap::new();
+
+        // We specifically inject an "unparsed" attribute right before this call
+        // so that we can pluck it out here since `try_from` only takes a single argument
+        let original = attrs
+            .remove("unparsed")
+            .ok_or_else(|| UrlParseError::MissingAttribute("unparsed"))?
+            .to_string()  // The `to_string` impl for Value adds extra quotes
+            .trim_matches('\"')
+            .to_string();
```

Comment:
Instead of trimming `"` left and right, why not just unwrap the string value instead?
I find that easier to follow through than formatting as string, manipulating the string and cloning it again.

```suggestion
            .as_str()
            .unwrap()  // This attribute is known to be a string 
            .to_string();
```